virttest/qemu_devices/qcontainer.py

```diff
@@ -917,7 +920,7 @@ def idx_of_next_named_bus(self, bus_pattern):
     def cleanup_daemons(self):
         """Clean up daemons."""
         for dev in self:
-            if isinstance(dev, qdevices.QVirtioFSDev):
+            if isinstance(dev, (qdevices.QVirtioFSDev, qdevices.QswtpmDev)):
```

Comment:
My point is, check the key property of such type devices that will start a daemon, so we don't need to append specific device type each time add new one.

Reply:
Hi, @vivianQizhu 
okay, I got your point. but I think that `cleanup_daemons` is a higher logical expression, should not check the device whether has `_stop_daemon` property and if there is such a scenario that the `qedvice`  instance just starts a daemon while doesn't stop it due to its behaviors, here `cleanup_daemons` will stop it by default.
`if isinstance(dev, (qdevices.QVirtioFSDev, qdevices.QswtpmDev)):`  like this way, will be more secure.

@luckyh what's your idea about @vivianQizhu 's suggestion?

Reply:
There are 2 (or maybe 3) approaches I can come up with:

1. checking if the dev has the `clear` method. As you said, it is improper while some devs may use `clear` for other usages, however, that may not be a good design since it makes people confusing to the `clear` method of different dev classes. So overall, it is a possible way to me.

2. introducing a new base class and checking the relationship for devs, for example:
```python
class QDaemonDev(QBaseDevice):
    ...

class QTPMDev(QDaemonDev):
    ...

if isinstance(dev, qdevices.QDaemonDev):
    ...
```

3. introducing a new class property as a flag and checking whether it is set. This is a variant of the second approach, and to me it is not good as that.


I'd prefer the second approach since it makes more sense to me, but all the approaches are acceptable, thanks.

Reply:
Hi @luckyh 
Thanks for your suggestions.
@vivianQizhu How about you?

Reply:
Both okay for me and prefer the second solution too. Thanks.

Reply:
Hi @vivianQizhu 
Have updated it by the second solution.
Thanks.